master/buildbot/test/unit/test_status_web_status_jsontestresults.py

```diff
@@ -26,16 +26,24 @@
 
 class TestJSONTestResource(unittest.TestCase):
     def setupStatus(self):
+        self.artifactServerPath = "/test/artifact/server/path"
+        def getProperty(x, y):
+            if x == "artifactServerPath" and y == "":
+                return self.artifactServerPath
+            return None
+
         st = self.build_step_status = mock.Mock(BuildStepStatus)
 
         build_status = mock.Mock(BuildStatus)
         build_status.getNumber = lambda: 1
+
         builder_status = mock.Mock(BuilderStatus)
         builder_status.getFriendlyName = lambda: "BuilderStatusFriendlyName"
```

Comment:
By the way, all of these could also be written as:

`build_status.getFriendlyName.return_value = "BuilderStatusFriendlyName"`